### PR TITLE
adjust plumbing to USE cable_surface_types offline

### LINKED ISSUE #436: `#ifdef` okay for now - but surely (for this one) we can get round the need for this by modifying the files names in the offline/coupled codebases to be the same.
`#ifdef` okay for now - but surely (for this one) we can get round the need for this by modifying the files names in the offline/coupled codebases to be the same.

_Originally posted by @har917 in https://github.com/CABLE-LSM/CABLE/pull/430#discussion_r1816356121_
            
I thought I had raised an issue about this, but at a glance it doesnt look like it. Yes change names, as first step, ultimately read surface_types from namelist as in JAC, but as with other use as well - offline doesnt have access to same JULES checks/balances, as well as other stuff. Should at least combine *2 coupled versions and sit with a coupled and an offline version of namelist reader.

### PR DESCRIPTION
**Pull request by JhanSrbinovsky**
# CABLE

The same module name can indeed be USEd as there are seperate versions picked up by coupled/offline apps. However some additional plumbing needs to be tweaked offline.

Please see issue for further details:
Fixes #436
## Checklist

- [x] The new content is accessible and located in the appropriate section.
- [x] I have checked that links are valid and point to the intended content.
- [x] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

----
📚 Documentation preview 📚: https://cable--446.org.readthedocs.build/en/446/

### CODE REVIEW

**Comment by JhanSrbinovsky** [comment]
@har917 you're tagged in the issue but perhaps here is more appropriate

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,6 +67,7 @@ add_library(
     src/offline/cbl_model_driver_offline.F90
     src/offline/landuse_inout.F90
     src/offline/spincasacnp.F90
+    src/offline/cable_surface_types.F90
```
Comment (bschroeter) [inline CMakeLists.txt:70]: There was an indent issue here, fixed it.

**Comment by JhanSrbinovsky** [comment]
@bschroeter can you have a look at this. The next tranche of AM3 updates will be much simpler if this does in first

**Comment by JhanSrbinovsky** [comment]
> @bschroeter can you have a look at this. The next tranche of AM3 updates will be much simpler if this does in first

 Actually to get ESM1,6 to work as well - so Labels etc will be updated

**Review by ccarouge — APPROVED** [review]
It's fine for me. I've made a commit (so pull before pushing if needed) to align all the USE statements where it was obviously needed from the changes view. 

Is there a plan to change `cable_surface_types.F90` to a namelist down the track?

**Comment by JhanSrbinovsky** [comment]
> Is there a plan to change `cable_surface_types.F90` to a namelist down the track?

absolutely

**Pull request merged**